src/library/parserm3u.cpp

```diff
@@ -107,10 +107,24 @@ bool ParserM3u::writeM3UFile(const QString &file_str, const QList<QString> &item
     QString fileContents(QStringLiteral("#EXTM3U\n"));
     for (const QString& item : items) {
         fileContents += QStringLiteral("#EXTINF\n");
-        if (useRelativePath) {
-            fileContents += baseDirectory.relativeFilePath(item) + QStringLiteral("\n");
+        QByteArray trackByteArray = QTextCodec::codecForName(kStandardM3uTextEncoding)
+                                            ->fromUnicode(item);
+        if ((trackByteArray == item) || useUtf8) {
+            if (useRelativePath) {
+                fileContents += baseDirectory.relativeFilePath(item) + QStringLiteral("\n");
+            } else {
+                fileContents += item + QStringLiteral("\n");
+            }
         } else {
-            fileContents += item + QStringLiteral("\n");
+            if (useRelativePath) {
+                QUrl itemRelativeUrl = QUrl::fromLocalFile(baseDirectory.relativeFilePath(item));
```

Comment:
I am not sure if this is working. For my understanding file URLs are always absolute. 
I would just delete the useRelativePath branch and add a source code comment about the issue.

Reply:
I am not aware of relative paths in URLs
You may verify it with VLC player wether it excepts such paths, or reimporting with Mixxx. If it turns out that it works flawlessly we can consider to add a solution. In general I prefer to stick with standards.

Reply:
I have tried to re import relative paths on Mixxx and it worked. But it is not working on VLC. It was giving an error for the tracks which has missing characters for all URL located tracks(error is called "VLC is unable to open the MRL").

VLC is also giving that error in normal way when there are missing characters aswell.

I think as you said it is better to stick with standards. So, I will leave there a source code comment such as //URL Location is not working for Relative Paths